tests/archives/dns_archive_test.py

```diff
@@ -0,0 +1,67 @@
+"Tests for the DNSArchive."
+
+import numpy as np
+
+from ribs.archives import DNSArchive
+
+
+def test_add_from_empty():
+    archive = DNSArchive(
+        solution_dim=2,
+        measure_dim=2,
+        capacity=10,
+        k_neighbors=5,
+    )
+    assert archive.empty, "Archive should be empty upon creation"
+
+    archive.add(
+        np.array([[0, 0], [1, 1]]),
+        np.array([0, 1]),
+        np.array([[0, 0], [1, 1]]),
+    )
+
+    # now there should be two solutions in the archive
+    assert len(archive) == 2, "Archive should have two solutions"
+
+    # now add 10 more solutions, cap should stay at 10
+    archive.add(
+        np.random.Generator(np.random.PCG64()).random((10, 2)),
```

Comment:
At the start of this test function, create a random generator with `rng = np.random.default_rng(42)`, where 42 is your seed. Then, call `rng.random()` in this code.